### DISCUSSION TITLE
Multi-Field index

### DISCUSSION THREAD

**Discussion by natgross** [cofu-app/cbl-dart#384] [Q&A] [answered]
Does cbl-dart support multi-field index? Suppose I want to index on lastName, then firstName in one index.
What would be the syntax for this? (The createIndex api allows only for one string property name.)
Also for ascending/descending per property.
Thanks.

**Comment by blaugold** [comment, marked as answer]
Yes, multi-field indexes are supported. You have two options for the syntax to specify an index: N1QL or query builder expressions.

```dart
// Query builder syntax
database.createIndex('User:Name', ValueIndex([
  ValueIndexItem.property('firstName'),
  ValueIndexItem.property('lastName'),
]));

// N1QL syntax
database.createIndex('User:Name', ValueIndexConfiguration([
  'firstName',
  'lastName',
]));
```

I don't think that it's possible to specify a sort direction for indexes, but I'll look into that.

**Reply by blaugold** [reply]
I did not find anything on specifying the order of indexes in the Couchbase Lite docs, so I assume it's not supported. The first field in a multi index should be iterable in both directions, though. If it turns out that this is a feature that you need, the [Couchbase forum](https://forums.couchbase.com/) would be the best place to create a feature request.
